operators/src/processing/temporal_raster_aggregation.rs

```diff
@@ -0,0 +1,1074 @@
+                return value;
+            }
+        }
+
+        if acc < value {
+            acc
+        } else {
+            value
+        }
+    }
+}
+
+struct MaxAccFunction {}
+
+impl AccFunction for MaxAccFunction {
+    fn acc<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T {
+        if let Some(no_data) = no_data {
+            if acc == no_data || value == no_data {
+                return no_data;
+            }
+        }
+
+        if acc > value {
+            acc
+        } else {
+            value
+        }
+    }
+}
+
+struct MaxIgnoreNoDataAccFunction {}
+
+impl NoDataIgnoringAccFunction for MaxIgnoreNoDataAccFunction {
+    fn acc_ignore_no_data<T: Pixel>(no_data: Option<T>, acc: T, value: T) -> T {
+        if let Some(no_data) = no_data {
+            if value == no_data {
+                return acc;
+            } else if acc == no_data {
+                return value;
+            }
+        }
+
+        if acc > value {
+            acc
+        } else {
+            value
+        }
+    }
+}
+
+fn fold_fn<T, C>(acc: (RasterTile2D<T>, ()), tile: RasterTile2D<T>) -> (RasterTile2D<T>, ())
+where
+    T: Pixel,
+    C: AccFunction,
+{
+    let mut acc = acc.0;
+    let grid = match (acc.grid_array, tile.grid_array) {
+        (GridOrEmpty::Grid(mut a), GridOrEmpty::Grid(g)) => {
+            a.data = a
+                .inner_ref()
+                .iter()
+                .zip(g.inner_ref())
+                .map(|(x, y)| C::acc(a.no_data_value, *x, *y))
+                .collect();
+            GridOrEmpty::Grid(a)
+        }
+        (GridOrEmpty::Empty(e), _) | (_, GridOrEmpty::Empty(e)) => GridOrEmpty::Empty(e),
+    };
+
+    acc.grid_array = grid;
+    (acc, ())
+}
+
+fn no_data_ignoring_fold_fn<T, C>(
+    acc: (RasterTile2D<T>, ()),
+    tile: RasterTile2D<T>,
+) -> (RasterTile2D<T>, ())
+where
+    T: Pixel,
+    C: NoDataIgnoringAccFunction,
+{
+    let mut acc = acc.0;
+    let grid = match (acc.grid_array, tile.grid_array) {
+        (GridOrEmpty::Grid(mut a), GridOrEmpty::Grid(g)) => {
+            a.data = a
+                .inner_ref()
+                .iter()
+                .zip(g.inner_ref())
+                .map(|(x, y)| C::acc_ignore_no_data(a.no_data_value, *x, *y))
+                .collect();
+            GridOrEmpty::Grid(a)
+        }
+        (GridOrEmpty::Grid(a), GridOrEmpty::Empty(_)) => GridOrEmpty::Grid(a),
+        (GridOrEmpty::Empty(_), GridOrEmpty::Grid(g)) => GridOrEmpty::Grid(g),
+        (GridOrEmpty::Empty(a), GridOrEmpty::Empty(_)) => GridOrEmpty::Empty(a),
+    };
+
+    acc.grid_array = grid;
+    (acc, ())
+}
+
+pub fn fold_future<T, C>(
+    accu: (RasterTile2D<T>, ()),
+    tile: RasterTile2D<T>,
+) -> impl Future<Output = Result<(RasterTile2D<T>, ())>>
+where
+    T: Pixel,
+    C: AccFunction,
+{
+    tokio::task::spawn_blocking(|| fold_fn::<T, C>(accu, tile)).then(async move |x| match x {
+        Ok(r) => Ok(r),
+        Err(e) => Err(e.into()),
+    })
```

Comment:
Why is it necessary to spawn tasks? Couldn't you - as you once told me - just say that the method is async in order to get a future?

Reply:
0d2561a7b8cc2e6b8b3690a7ce2b9d6cbfa8ae24

Reply:
i do not think that this is correct. We discussed this and  need to find a solution.

Reply:
f8880760839c283a01ce233a024c663ed914dd0e